### PR TITLE
feat: add indicators for right buttons

### PR DESCRIPTION
**Pull request by olivierwilkinson**
| Status  | Ticket/Issue |
| :---: | :--: |
| Ready | [OS-1210](https://pi-top.atlassian.net/browse/OS-1210) |

Please review [OS-1209](https://github.com/pi-top/pi-top-4-Miniscreen/pull/212) first

#### Main changes
Add a gutter to the right of the HUDTileGroup with indicators for the
buttons on the right hand side of the miniscreen.

The indicators show that the select button can navigate into a menu with
a right arrow, or can perform an action with a tick. The cancel button
has a left arrow next to it when it's possible to navigate out of a
sub-menu.

Change the indicators when the current page of the top most
PaginatedTile changes, or when navigating in or out of a sub-menu.

Add events for when the PaginatedTile's scrolling animation starts or
finishes. This allows the indicators to be changed at the end of the
scroll animation.

Update the overview, battery, cpu, action and network pages to fit the
new page size.

Remove the icons from the action pages since they are not resizable and
could not fit with the more important text easily.

Refactor the battery page so that it is easier to work with.

### CODE REVIEW

**Comment by angusjfw** [comment]
battery pages not updating:
```
Feb 16 12:09:07 pi-top systemd[1]: Started pi-top OLED system menu.
Feb 16 12:09:12 pi-top pt-miniscreen[21630]: Starting main loop...
Feb 16 12:09:34 pi-top pt-miniscreen[21630]: Garbage collect Hotspot: <pt_miniscreen.hotspots.templates.image.Hotspot object at 0xa75e9a48>
Feb 16 12:09:37 pi-top pt-miniscreen[21630]: Exception in thread Thread-3:
Feb 16 12:09:37 pi-top pt-miniscreen[21630]: Traceback (most recent call last):
Feb 16 12:09:37 pi-top pt-miniscreen[21630]:   File "/usr/lib/python3.9/threading.py", line 954, in _bootstrap_inner
Feb 16 12:09:37 pi-top pt-miniscreen[21630]: Exception in thread Thread-2:
Feb 16 12:09:37 pi-top pt-miniscreen[21630]: Traceback (most recent call last):
Feb 16 12:09:37 pi-top pt-miniscreen[21630]:   File "/usr/lib/python3.9/threading.py", line 954, in _bootstrap_inner
Feb 16 12:09:37 pi-top pt-miniscreen[21630]:     self.run()
Feb 16 12:09:37 pi-top pt-miniscreen[21630]:   File "/usr/lib/python3.9/threading.py", line 892, in run
Feb 16 12:09:37 pi-top pt-miniscreen[21630]:     self._target(*self._args, **self._kwargs)
Feb 16 12:09:37 pi-top pt-miniscreen[21630]:     self.run()
Feb 16 12:09:37 pi-top pt-miniscreen[21630]:   File "/usr/lib/python3.9/threading.py", line 892, in run
Feb 16 12:09:37 pi-top pt-miniscreen[21630]:   File "/usr/lib/python3/dist-packages/pitop/common/ptdm.py", line 457, in __thread_method
Feb 16 12:09:37 pi-top pt-miniscreen[21630]:     self._target(*self._args, **self._kwargs)
Feb 16 12:09:37 pi-top pt-miniscreen[21630]:   File "/usr/lib/python3/dist-packages/pitop/common/ptdm.py", line 457, in __thread_method
Feb 16 12:09:37 pi-top pt-miniscreen[21630]:         self.invoke_callback_func_if_exists(self.invoke_callback_func_if_exists(
Feb 16 12:09:37 pi-top pt-miniscreen[21630]:   File "/usr/lib/python3/dist-packages/pitop/common/ptdm.py", line 474, in invoke_callback_func_if_exists
Feb 16 12:09:37 pi-top pt-miniscreen[21630]:     func(params)
Feb 16 12:09:37 pi-top pt-miniscreen[21630]:   File "/usr/lib/python3/dist-packages/pitop/battery/battery.py", line 40, in on_state_changed
Feb 16 12:09:37 pi-top pt-miniscreen[21630]:     self.on_capacity_change(capacity)  File "/usr/lib/python3/dist-packages/pitop/common/ptdm.py", line 474, in invoke_callback_func_if_exists
Feb 16 12:09:37 pi-top pt-miniscreen[21630]: TypeError: update_hotspots_properties() takes 1 positional argument but 2 were given
Feb 16 12:09:37 pi-top pt-miniscreen[21630]:     func(params)
Feb 16 12:09:37 pi-top pt-miniscreen[21630]:   File "/usr/lib/python3/dist-packages/pitop/battery/battery.py", line 40, in on_state_changed
Feb 16 12:09:37 pi-top pt-miniscreen[21630]:     self.on_capacity_change(capacity)
Feb 16 12:09:37 pi-top pt-miniscreen[21630]: TypeError: update_hotspots_properties() takes 1 positional argument but 2 were given
Feb 16 12:09:37 pi-top pt-miniscreen[21630]: Garbage collect Tile: <pt_miniscreen.tiles.menu.system.SystemMenuTile object at 0xa75e9f28>
```

File: pt_miniscreen/pages/hud/overview.py [hunk 1]
```diff
@@ -14,95 +14,100 @@
...
+                    self.offset_pos_for_vertical_center(CAPACITY_SIZE[1]),
+                ),
+            ),
+            HotspotInstance(self.text_hotspot, (TEXT_LEFT, TEXT_TOP)),
         ]
 
-        self.update_hotspots_properties()
-        self.setup_events()
+        # setup battery callbacks
+        battery.on_capacity_change = self.update_hotspots_properties
```
Comment (angusjfw) [inline pt_miniscreen/pages/hud/overview.py:99]:
```suggestion
        battery.on_capacity_change = lambda _: self.update_hotspots_properties()
```

**Comment by angusjfw** [comment]
also it would be nice to change the garbage collection logs to debug level

File: pt_miniscreen/pages/hud/overview.py (hunk 1 above)
Comment (angusjfw) [reply pt_miniscreen/pages/hud/overview.py:99]: i've tested this for all the events except 'full'

File: pt_miniscreen/pages/system/battery.py [hunk 2]
```diff
@@ -11,104 +11,100 @@
...
+                    self.offset_pos_for_vertical_center(CAPACITY_SIZE[1]),
+                ),
+            ),
+            HotspotInstance(self.text_hotspot, (TEXT_LEFT, TEXT_TOP)),
         ]
 
-        self.update_hotspots_properties()
-        self.setup_events()
+        # setup battery callbacks
+        battery.on_capacity_change = self.update_hotspots_properties
```
Comment (angusjfw) [inline pt_miniscreen/pages/system/battery.py:96]:
```suggestion
        battery.on_capacity_change = lambda _: self.update_hotspots_properties()
```

**Comment by jcapona** [comment]
after testing this with the changes @angusjfw suggested, everything seems to be working fine! I was able to test the 'full' state of the battery and the icon is updated as expected

**Comment by olivierwilkinson** [comment]
I've added in the changes Angus suggested after testing them while I rebased on top of master. I haven't changed anything else!

**Review by angusjfw — APPROVED** [review]

**Pull request merged**